### PR TITLE
Update segments documentation page

### PR DESCRIPTION
**Pull request by spxiwh**
The multi-ifo workflow will soon be moved over to the new segments format. As requested by @tdent, we need to improve the documentation to show how to use this new module. This patch contains the updated documentation, as discussed in SLACK. An example of the new page is here:

https://ldas-jobs.ligo.caltech.edu/~spxiwh/TMP_DOCS/latest/html/workflow/segments.html

but has the same content as this wiki:

https://github.com/spxiwh/pycbc/wiki/New-DQ-ini-syntax

### CODE REVIEW

File: docs/workflow/segments.rst [hunk 1]
```diff
@@ -35,19 +35,150 @@ Using this module requires a number of things
...
+describe the segment options
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Description of the ini file contents
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are three important segments options that we need to provide:
+
+* `segments-science`: This decides what times will be analysed to produce triggers. All times in this might be used to compute PSDs used in the final results. In previous runs this is normally the science flag with CAT_1 vetoes subtracted from it.
+* `segments-vetoes`: This decides what times will be vetoed when producing final significance. These times *are* analysed but are discarded after combining single-detector triggers together. In previous runs this is CAT_2 vetoes and hardware injections.
```
Comment (tdent) [inline docs/workflow/segments.rst]: Prefer 'final candidate event lists'. Last sentence : 'Normally the time discarded comprises times vetoed at CAT_2 and times of hardware injections'

File: docs/workflow/segments.rst [hunk 2]
```diff
@@ -35,19 +35,150 @@ Using this module requires a number of things
...
+* `segments-veto-definer-url`: As previously, this is the location of the veto definer.
+
+Note that this obeys the usual workflow tagging rules. If you supply `segments-science` in `workflow-segments` it will be valid for all ifos. Or, if you want to supply different values for different ifos (e.g. because the Virgo SCIENCE flag is named differently to L1 and H1) you can use `workflow-segments-ifos`.
```
Comment (tdent) [inline docs/workflow/segments.rst]: Oh, I hadn't noticed that this difference existed. Is there a link for the usual rules for this case of adding 'ifos' to the tag?

Or do you mean workflow-segments-ifoname where ifoname is h1, v1 etc

File: docs/workflow/segments.rst [hunk 3]
```diff
@@ -35,19 +35,150 @@ Using this module requires a number of things
...
+
+The `segments-science` and `segments-vetoes` are provided as a *comma-separated* list of flags. We will document what these are below.
```
Comment (tdent) [inline docs/workflow/segments.rst]: 'documented below' will do

File: docs/workflow/segments.rst [hunk 4]
```diff
@@ -35,19 +35,150 @@ Using this module requires a number of things
...
+
+^^^^^^^^^^^^^^^^^^^^^
+Flag syntax
+^^^^^^^^^^^^^^^^^^^^^
+
+We've said that `segments-science` and `segments-vetoes` look something like
+
+`segments-science = FLAG_1,FLAG_2`
+
+So let's describe now what FLAG_1 might look like. There is a bit of subtlety to the flag syntax, so let's start with something basic, and then add possible complexity.
```
Comment (tdent) [inline docs/workflow/segments.rst]: These 2 sentences and the next one just say 'we start with a simple example of what can be given as the value of flag_1'
Don't see what the verbosity adds

File: docs/workflow/segments.rst [hunk 5]
```diff
@@ -35,19 +35,150 @@ Using this module requires a number of things
...
+
+At the very basic level you can set (for the value of FLAG_1):
+
+`+SCIENCE` or `-SCIENCE`
+
+In this case SCIENCE is the flag name. It is what will be queried from the segment server. The ifo will be automatically appended (so this becomes `${IFO}:SCIENCE` when querying). The `+` or `-` is important (and not optional). If the sign as a `+` then the flag is *added* to the list of times, if it is `-` then the flag is removed. All `+` flags are processed first and then all `-` flags are removed from that.
```
Comment (tdent) [inline docs/workflow/segments.rst]: Remove the emphasis on added, doesn't make the text clearer.

Remove all the brackets, they don't help readability

File: docs/workflow/segments.rst [hunk 6]
```diff
@@ -35,19 +35,150 @@ Using this module requires a number of things
...
+
+In `segments-science` `+` means that that time *should* be included in the analysis. A `-` means it should not be included.
```
Comment (tdent) [inline docs/workflow/segments.rst]: Remove emphasis on should

File: docs/workflow/segments.rst [hunk 7]
```diff
@@ -35,19 +35,150 @@ Using this module requires a number of things
...
+
+In `segments-vetoes` `+` means that that time *should* be included in the vetoed times (and so *not* included in the analysis). A `-` means it should not be included in the vetoed times (so that time can be analysed) ... It is not envisaged that `-` flags will be used often in `segments-vetoes`.
```
Comment (tdent) [inline docs/workflow/segments.rst]: Remove apparently random emphases and unnecessary brackets. Full stop not 3 dots.

Is this really this way round? The example ini file does use - for the cat2 and hwinj vetoes. Is it wrong?

File: docs/workflow/segments.rst (hunk 2 above)
Comment (spxiwh) [reply docs/workflow/segments.rst]: I mean workflow-segments-ifoname where ifoname is h1, v1 etc, clarifying now.

File: docs/workflow/segments.rst (hunk 7 above)
Comment (spxiwh) [reply docs/workflow/segments.rst]: The example ini here is incorrect, the text is correct. Will fix that.

**Comment by spxiwh** [comment]
@tdent I think I've addressed your concerns.

File: docs/workflow/resources/example_dq.ini [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+[workflow-segments-h1]
+; NOTE: It's important to check the version number
+segments-science = +DMT-ANALYSIS_READY:1,-CAT_1
+
+[workflow-segments-l1]
+; NOTE: It's important to check the version number
+segments-science = +DMT-ANALYSIS_READY:1,-CAT_1
+
+[workflow-segments-v1]
```
Comment (tdent) [inline docs/workflow/resources/example_dq.ini]: since there is no other V1 entry in this file, remove this section?

Comment (tdent) [reply docs/workflow/resources/example_dq.ini]: I took it out, will merge soon unless there are any further changes/objections

**Pull request merged**